src/pages/apps/[id]/settings/_components/FormAppSettings.tsx

```diff
@@ -68,6 +68,7 @@ const FormAppSettings: React.FC<Props> = ({ api, app, additionalSettings }) => {
       </Form.Section>
       <Form.Section label="Repository">
         <Form.Input
+          key={toRepoAddr(app.repo)}
```

Comment:
Would using the `value` property instead of `defaultValue` work here? Maybe we won't need the `key` hack in that case? WDYT